src/utils.py

```diff
@@ -189,6 +189,10 @@ def get_env() -> dict[str, str]:
 
 def update_env(env: dict[str, str]) -> None:
     """Updates /etc/environment file."""
-    updated_env = get_env() | env
-    content = "\n".join([f"{key}={value}" for key, value in updated_env.items()])
+    current_env = get_env()
+    if not env or current_env == env:
+        return
+
```

Comment:
**question:** What does this solve? Avoiding re-writing the same content to a file?
If so, I'd suggest removing these lines, as there's no downside to writing a file every time, and it makes the code more concise.

Reply:
Actually, I've been addressing a comment from @zmraul , [here](https://github.com/canonical/kafka-operator/pull/159#discussion_r1427929488). I don't mind that much, but I also believe it is a small improvements worth having at this point (I don't think there was any downside on having this, and we actually touch the file only when it is necessary, providing a bit better description when inspecting the files)